tests/testthat/test-read.R

```diff
@@ -74,6 +75,61 @@ test_that("Reading H5AD works with native reader", {
     expect_identical(colnames(colData(sce)), "cell_type")
 })
 
+test_that("Reading v0.8 H5AD works with native reader", {
+    sce_py <- readH5AD(file_v08)
+    sce_r <- readH5AD(file_v08, reader='R')
+
+    expect_identical(rownames(sce_py), rownames(sce_r))
+    expect_identical(colnames(sce_py), colnames(sce_r))
+
+    # rowData is just an empty DataFrame here
+    expect_identical(rowData(sce_py), rowData(sce_r))
+
+    # colData colnames not in same order, but check the sets are the same
+    expect_true(setequal(colnames(colData(sce_py)), colnames(colData(sce_r))))
+
+    # check colData columns that Python reader is able to handle
+    good_coldat_columns <- c('cell_type', 'dummy_bool', 'dummy_int',
+                             'dummy_num', 'dummy_num2')
+
+    expect_equal(colData(sce_py)[,good_coldat_columns],
+                 colData(sce_r)[,good_coldat_columns])
+
+    # Manually check colData columns that Python reader doesn't handle
+    # right (it reads them as environment objects)
+    expect_equal(colData(sce_r)$dummy_bool2,
+                 c(FALSE, NA, rep(TRUE, 638)))
+
+    expect_equal(colData(sce_r)$dummy_int2,
+                 c(NA, rep(42, 639)))
+
+    # check the X assay
+    expect_identical(assays(sce_py), assays(sce_r))
+
+    # check the easy metadata columns
+    for (key in c('highlights', 'iroot', 'dummy_int')) {
+        expect_equal(metadata(sce_py)[[key]], metadata(sce_r)[[key]])
+    }
+
+    # python reads uns[dummy_bool] as an array, so convert it (is that
+    # a bug in the python reader?)
+    expect_equal(
+        as.vector(metadata(sce_py)[['dummy_bool']]),
+        metadata(sce_r)[['dummy_bool']]
```

Comment:
Hmmmm...not sure. It may be something to do with how **{reticulate}** converts single values.